### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by AlselwiAisha**
**Description:**
- Add Redux Toolkit (npm install react-redux @reduxjs/toolkit).
- Create a directory that will contain all my Redux logic (/src/redux)
- Configure a Redux store (/src/redux/store.js)
- Define a slice of state for books that:
      - Includes an array of books (initial state: empty array)
      - Includes a reducer that adds a book
      - Includes a reducer that removes a book
- Define a slice of state for categories that:
      - Includes an array of categories (initial state: empty array)
      - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/redux/books/booksSlise.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState: {
+    Books: [],
+  },
+  reducers: {
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
```
Comment (Qoosim) [inline src/redux/books/booksSlise.js]: - [x] Great job adding reducer to your app. However, let's add litlte stuff here for proper implementation. Your payload is the book object you're adding and they must be added to the Books array. Check this snippet below to have an idea how it should work.
```
reducers: {
    addBook: (state, action) => {
      state.Books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },

```

File: src/redux/books/booksSlise.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    removeBook: (state, action) => {
+      const bookId = action.payload;
+      state.filter((book) => book.id !== bookId);
```
Comment (Qoosim) [inline src/redux/books/booksSlise.js]: - [x] Also this one. It should be this way.

```
   state.Books.filter((book) => book.id !== bookId)
 
   or 
   
   removeBook: (state, action) => {
      state.Books = state.books.filter((book) => book.id !== action.id)
    },
  
```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @AlselwiAisha, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @AlselwiAisha , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Implemented the changes requested to fix errors in the bookSlice.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**